provisioner/terraform/tfparse/tfparse.go

```diff
@@ -0,0 +1,490 @@
+			logger.Debug(ctx, "only .tf files can be parsed", "filename", dataResource.Pos.Filename)
+			continue
+		}
+		// We know in which HCL file is the data resource defined.
+		file, diags = parser.ParseHCLFile(dataResource.Pos.Filename)
+		if diags.HasErrors() {
+			return nil, xerrors.Errorf("can't parse the resource file: %s", diags.Error())
+		}
+
+		// Parse root to find "coder_workspace_tags".
+		content, _, diags := file.Body.PartialContent(rootTemplateSchema)
+		if diags.HasErrors() {
+			return nil, xerrors.Errorf("can't parse the resource file: %s", diags.Error())
+		}
+
+		// Iterate over blocks to locate the exact "coder_workspace_tags" data resource.
+		for _, block := range content.Blocks {
+			if !slices.Equal(block.Labels, []string{"coder_workspace_tags", dataResource.Name}) {
+				continue
+			}
+
+			// Parse "coder_workspace_tags" to find all key-value tags.
+			resContent, _, diags := block.Body.PartialContent(coderWorkspaceTagsSchema)
+			if diags.HasErrors() {
+				return nil, xerrors.Errorf(`can't parse the resource coder_workspace_tags: %s`, diags.Error())
+			}
+
+			if resContent == nil {
+				continue // workspace tags are not present
+			}
+
+			if _, ok := resContent.Attributes["tags"]; !ok {
+				return nil, xerrors.Errorf(`"tags" attribute is required by coder_workspace_tags`)
+			}
+
+			expr := resContent.Attributes["tags"].Expr
+			tagsExpr, ok := expr.(*hclsyntax.ObjectConsExpr)
+			if !ok {
+				return nil, xerrors.Errorf(`"tags" attribute is expected to be a key-value map`)
+			}
+
+			// Parse key-value entries in "coder_workspace_tags"
+			for _, tagItem := range tagsExpr.Items {
+				key, err := previewFileContent(tagItem.KeyExpr.Range())
+				if err != nil {
+					return nil, xerrors.Errorf("can't preview the resource file: %v", err)
+				}
+				key = strings.Trim(key, `"`)
+
+				value, err := previewFileContent(tagItem.ValueExpr.Range())
+				if err != nil {
+					return nil, xerrors.Errorf("can't preview the resource file: %v", err)
+				}
+
+				logger.Info(ctx, "workspace tag found", "key", key, "value", value)
+
+				if _, ok := workspaceTags[key]; ok {
+					return nil, xerrors.Errorf(`workspace tag %q is defined multiple times`, key)
+				}
+				workspaceTags[key] = value
+			}
+		}
+	}
+	return workspaceTags, nil
+}
+
+//	WorkspaceTagDefaultsFromFile extracts the default values for a `coder_workspace_tags` resource from the given
+//
+// file. It also ensures that any uses of the `coder_workspace_tags` data source only
+// reference the following data types:
+// 1. Static variables
+// 2. Template variables
+// 3. Coder parameters
+// Any other data types are not allowed, as their values cannot be known at
+// the time of template import.
+func WorkspaceTagDefaultsFromFile(ctx context.Context, logger slog.Logger, file []byte, mimetype string) (tags map[string]string, err error) {
+	module, cleanup, err := loadModuleFromFile(file, mimetype)
```

Comment:
Probably out of the scope of this PR: are we going to lookup files in other Terraform modules or nested directories?

Reply:
I'm not sure that's something we want to support right now. Allowing that would enable 'spooky action at a distance' for Terraform modules, leading to hard-to-debug issues with figuring out where workspace tags are actually defined. We can revisit at a later date.